src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -26,6 +26,11 @@ NonClientIslandWindow::NonClientIslandWindow(const ElementTheme& requestedTheme)
 {
 }
 
+NonClientIslandWindow::~NonClientIslandWindow()

```

Comment:
Just added redundancy? I don't think we've got any code that was removed that was calling `Close` before, right?

Reply:
Oh, it's terrible... This change fixes a bug and you might already know what I mean when I say: `IslandWindow::Close` is a virtual function which the `~IslandWindow` destructor calls. `NonClientIslandWindow` overloads `Close`. *boom*

But in my opinion this sorely belongs into the "only obvious in hindsight" category of arcane C++ footnukes (can't have enough of those in a language!), so let me explain in more detail:
* No matter whether you define a destructor or not, C++ will define it's own "real-destructor", because that one is responsible for destroying non-trival class members (like `std::string`s and such). So, while `NonClientIslandWindow` doesn't have an explicit destructor, it gets one anyways which virtually overloads the `IslandWindow` destructor.
* With inheritance, destructors are called in reverse starting from child classes up to parent classes. In other words, first `~NonClientIslandWindow` is called and then `~IslandWindow`.

In short, when `~IslandWindow` gets called and it tries to call the virtual function `Close`, the child class `NonClientIslandWindow` is already destroyed, which means that the vtable pointer now points to `IslandWindow`, which means that `IslandWindow::Close` is called and not `NonClientIslandWindow::Close`. The above change fixes this.